scipy/stats/_sobol.pyi

```diff
@@ -0,0 +1,52 @@
+import numpy.typing as npt
+from scipy._lib._util import IntNumber
+
+def initialize_v(
+    v : npt.ArrayLike, 
+    dim : IntNumber
+) -> None: ...
+
+def _cscramble (
+    dim : IntNumber,
+    ltm : npt.ArrayLike,
+    sv: npt.ArrayLike
+) -> None: ...
+
+def _fill_p_cumulative(
+    p: npt.ArrayLike,
+    p_cumulative: npt.ArrayLike
+) -> None: ...
+
+def _draw(
+    n : IntNumber,
+    num_gen: IntNumber,
+    dim: IntNumber,
+    sv: npt.ArrayLike,
+    quasi: npt.ArrayLike,
+    result: npt.ArrayLike
+    ) -> None: ...
+
+def _fast_forward(
+    n: IntNumber,
+    num_gen: IntNumber,
+    dim: IntNumber,
+    sv: npt.ArrayLike,
+    quasi: npt.ArrayLike
+    ) -> None: ...
+
+def _categorize(
+    draws: npt.ArrayLike,
+    p_cumulative: npt.ArrayLike,
+    result: npt.ArrayLike
+    ) -> None: ...
+
+def initialize_direction_numbers() -> None: ...
+
+_MAXDIM: int = ...
+_MAXBIT: int = ...
```

Comment:
```suggestion
from typing_extensions import Literal  # Preferrably place this import at the top of the file

_MAXDIM: Literal[21201]
_MAXBIT: Literal[30]
_MAXDEG: Literal[18]
```
There are two proposed changes here:
* First off, `_MAXDEG` currently seems to be missing from the stubs. https://github.com/scipy/scipy/blob/4ec6f29c0344ddce80845df0c2b8765fc8f27a72/scipy/stats/_sobol.pyx#L28-L30
* Secondly,  these three integers all have a constant pre-defined value, so we can use [literal](https://docs.python.org/3/library/typing.html#typing.Literal) integers here to provide just a bit more information.

Reply:
Ok for the literals, makes sense. But then I guess we should also do it in `_qmc.py` as these where defined as `ClassVar[int]`?

Reply:
That would definitely be an improvement, yes, though it is not necessarily mandatory. 
Namely, `ClassVar[int]` is still a super-type of `ClassVar[Literal[30]]`, so this change shouldn't introduce any new mypy errors.